vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/pool/SqlConnectionPool.java

```diff
@@ -145,18 +156,33 @@ public int size() {
     return pool.size();
   }
 
-  public void evict() {
+  public void evict(long connectionTimeout) {
     long now = System.currentTimeMillis();
     pool.evict(conn -> conn.shouldEvict(now), ar -> {
       if (ar.succeeded()) {
         List<PooledConnection> res = ar.result();
         for (PooledConnection conn : res) {
           conn.close(Promise.promise());
         }
+        startMin(connectionTimeout);
       }
     });
   }
 
+  public void startMin(long connectionTimeout) {
+    int needed = max(minSize - pool.size(), 0);
+    if (needed > 0) {
+      ContextInternal context = vertx.getOrCreateContext();
+      for (int i = 0; i < needed; ++i) {
+        acquire(context, connectionTimeout, (ar) -> {
```

Comment:
I'm wondering if there should be a flavor of `acquire` that forces new connection creation. Thinking of the following scenario: 

If the `maxPoolSize` is 16, `minSize` is 8, and let's say steady state connection usage is 4 (so `pool.size()` will be approximately 4). In this case, is it possible that some/all `needed` calls to `acquire` (i.e. needed = 8 - 4 = 4) end up getting one or more of the steady state connections when they are not in use, and as a result,  < 4 (or even 0) new connections get created, and `pool.size()` doesn't reach close to 8 (or worse, remains at 4)?

@tsegismont  is this a valid concern?

Reply:
Over time (e.g a period of 10 - 30 seconds) this brings up the required number of connections and keeps it there.

Even though I currently don't see this behavior your example and concern may be valid. There is a simpler fix though, just pas the required minimum to the evict function; it can easily stop the eviction test when the minimum would be breached.

Reply:
This requires a change to vert.x core.

Reply:
I've updated the PR with a simpler fix. Instead of calculating a needed value, it now checks if the pool is "low" and acquires `minSize` connections immediately (in parallel).

This seems to work pretty well and ensures that `minSize` will be honored at all times. The downside is that it _may_ start more than `minSize connections depending on if any of the connections are in use.

In practice this "downside" doesn't appear that often. Remember that to trigger the acquisition of connections the pool has to be in some kind of "idle" state because connections are idling and falling below the pool minimum. Worst case a few extra connections are activated for a short period of before idling away.

In the end the implementation is somewhere between a `minSize` and `minAvailable` depending on the state of the pool and I'm pretty ok with that.

One thing to remember when discussing this is that the pool does wait a beat to see if a connection becomes available; it doesn't immediately start a connection when an acquisition request cannot be immediately fulfilled.

Looking back to your scenario of current pool size of `4` and a `minSize` of `8`. The pool wold attempt to acquire 8 connections. In the case where all 4 connections are in-use, the pool would attempt to acquire 8 connections, potentially bumping it to 12 connections; but only if none of the connections are recycled while the pool is acquiring the connections.

Reply:
Thanks, I think this should work for most cases.

Reply:
Because of the asynchronous nature of `acquire`, we shall wait for all operations to complete before recycling the lease. Otherwise, there is no guarantee that some of the operations will complete before all operations are submitted.